### PR TITLE
Add CHARINDEX function for sqlserver

### PR DESCRIPTION
**Pull request by eukleshnin**
Added CHARINDEX function support per #1059. Only supported for sql server for now, will add other dialects in the next PR.

### CODE REVIEW

File: exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/Function.kt [hunk 1]
```diff
@@ -137,6 +137,14 @@ class Trim<T : String?>(
     override fun toQueryBuilder(queryBuilder: QueryBuilder): Unit = queryBuilder { append("TRIM(", expr, ")") }
 }
 
+/**
+ * Represents an SQL function that returns the index of the first occurrence of [char] in [expr] or 0
+ */
+class CharIndex<T : String?>(val expr: Expression<T>, val char: String) : Function<Int>(IntegerColumnType()) {
```
Comment (AlexeySoshin) [inline exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/Function.kt:143]: There's a more generic function called Position in PostgreSQL and MySQL
https://www.postgresql.org/docs/15/functions-string.html
https://dev.mysql.com/doc/refman/8.0/en/string-functions.html

I wonder if calling this `Position` and implementing it as `CharIndex` only for MSSQL would be a better option here?

Comment (eukleshnin) [reply exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/Function.kt:143]: Thanks for pointing that out!

Yeah, that makes sense. `CharIndex` seems to be specific to SQL server, while `Position` is used in 2 systems. Though I wonder if it even should be `Locate`, as it's used in MySQL, H2 and Maria DB.

Comment (AlexeySoshin) [reply exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/Function.kt:143]: I agree. It seems that `LOCATE()` is the most generic, as it has also the third parameter of start position, that can be defaulted.

Comment (eukleshnin) [reply exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/Function.kt]: I renamed the function and added implementation for all supported dialects.
I wonder if all dialects are tested on PRs? If not how can I run tests for the remaining ones on CI?

**Review by AlexeySoshin — APPROVED** [review]

**Review by bog-walk — APPROVED** [review]
@eukleshnin  Thank you for submitting this PR.

**Review by e5l — APPROVED** [review]

**Pull request merged**